File: legacy/enrich_data.py

import json
import math
import re

def haversine(lat1, lon1, lat2, lon2):
    R = 6371  # Earth radius in km
    phi1, phi2 = math.radians(lat1), math.radians(lat2)
    dphi = math.radians(lat2 - lat1)
    dlambda = math.radians(lon2 - lon1)
    a = math.sin(dphi / 2)**2 + math.cos(phi1) * math.cos(phi2) * math.sin(dlambda / 2)**2
    return 2 * R * math.asin(math.sqrt(a))

# Coordinates
BASES = {
    "altoParaiso": (-14.1311, -47.5147),
    "saoJorge": (-14.1722, -47.8183),
    "cavalcante": (-13.7915, -47.4578)
}

def calculate_road_distance(lat1, lon1, lat2, lon2):
    # For Chapada, road distance is roughly 1.3x to 1.8x the straight line distance
    # especially for dirt roads.
    d = haversine(lat1, lon1, lat2, lon2)
    return round(d * 1.5) # Average multiplier for mountain roads

def get_google_photos_url(name):
    query = f"{name} Chapada dos Veadeiros".replace(" ", "+")
    return f"https://www.google.com/search?q={query}+fotos&tbm=isch"

def process_file():
    path = r'c:\Users\jaime\OneDrive\Desktop\aranumlabs-estetica\chapada-veadeiros\data.js'
    with open(path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Extract ATTRACTIONS array using regex (very basic)
    match = re.search(r'const ATTRACTIONS = (\[.*\]);', content, re.DOTALL)
    if not match:
        print("Could not find ATTRACTIONS array")
        return

    attractions_str = match.group(1)
    
    # We need to handle the fact that it might not be perfect JSON (it's JS)
    # But for our current data.js, it's mostly JSON-like after my previous refactor.
    try:
        # Pre-process to make it JSON-compatible if needed (replace single quotes etc)
        # Actually our current data.js is quite clean JSON-like inside the array.
        attractions = json.loads(attractions_str)
    except json.JSONDecodeError as e:
        print(f"Error parsing ATTRACTIONS: {e}")
        return

    for attr in attractions:
        lat = attr.get('lat')
        lng = attr.get('lng')
        
        if lat and lng:
            attr['distances'] = {
                "altoParaiso": calculate_road_distance(lat, lng, *BASES["altoParaiso"]),
                "saoJorge": calculate_road_distance(lat, lng, *BASES["saoJorge"]),
                "cavalcante": calculate_road_distance(lat, lng, *BASES["cavalcante"])
            }
        
        # Ensure googlePhotosUrl is set
        attr['googlePhotosUrl'] = get_google_photos_url(attr['name'])
        
        # Special case for Jardim de Maytrea as per user data
        if attr['id'] == 'jardim-maytrea':
            attr['distances'] = {
                "altoParaiso": 22,
                "saoJorge": 17,
                "cavalcante": 109
            }

    # Replace the array in the original content
    new_attractions_str = json.dumps(attractions, indent=4, ensure_ascii=False)
    new_content = re.sub(r'const ATTRACTIONS = \[.*\];', f'const ATTRACTIONS = {new_attractions_str};', content, flags=re.DOTALL)

    with open(path, 'w', encoding='utf-8') as f:
        f.write(new_content)
    
    print("Successfully updated distancing and photo links for all attractions.")

if __name__ == "__main__":
    process_file()

File: legacy/enrich_data.js

const fs = require('fs');
const path = require('path');

function haversine(lat1, lon1, lat2, lon2) {
    const R = 6371; // Earth radius in km
    const dLat = (lat2 - lat1) * Math.PI / 180;
    const dLon = (lon2 - lon1) * Math.PI / 180;
    const a = Math.sin(dLat / 2) * Math.sin(dLat / 2) +
        Math.cos(lat1 * Math.PI / 180) * Math.cos(lat2 * Math.PI / 180) *
        Math.sin(dLon / 2) * Math.sin(dLon / 2);
    const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
    return R * c;
}

const BASES = {
    altoParaiso: { lat: -14.1311, lng: -47.5147 },
    saoJorge: { lat: -14.1722, lng: -47.8183 },
    cavalcante: { lat: -13.7915, lng: -47.4578 }
};

function calculateRoadDistance(lat1, lon1, lat2, lon2) {
    const d = haversine(lat1, lon1, lat2, lon2);
    return Math.round(d * 1.5);
}

function getGooglePhotosUrl(name) {
    const query = name.replace(/ /g, '+');
    return `https://www.google.com/search?q=${query}+fotos+chapada+dos+veadeiros&tbm=isch`;
}

const dataPath = path.join(__dirname, 'data.js');
let content = fs.readFileSync(dataPath, 'utf8');

// Use a more robust regex for the ATTRACTIONS array
const match = content.match(/const ATTRACTIONS = (\[[\s\S]*?\]);/);
if (!match) {
    console.error("Could not find ATTRACTIONS array");
    process.exit(1);
}

// We need to handle the fact that it's JS, not strict JSON. 
// But given the previous refactor, it's very close to JSON.
let attractions;
try {
    // Attempt to evaluate the array literal safely
    // Since we know the content and it's our own structured data
    attractions = JSON.parse(match[1]);
} catch (e) {
    console.error("Error parsing ATTRACTIONS as JSON, trying simplified eval");
    try {
        // Fallback for non-strict JSON (like missing quotes on keys if they were there)
        attractions = eval(match[1]);
    } catch (e2) {
        console.error("Fatal: Could not parse ATTRACTIONS array", e2);
        process.exit(1);
    }
}

attractions.forEach(attr => {
    if (attr.lat && attr.lng) {
        attr.distances = {
            altoParaiso: calculateRoadDistance(attr.lat, attr.lng, BASES.altoParaiso.lat, BASES.altoParaiso.lng),
            saoJorge: calculateRoadDistance(attr.lat, attr.lng, BASES.saoJorge.lat, BASES.saoJorge.lng),
            cavalcante: calculateRoadDistance(attr.lat, attr.lng, BASES.cavalcante.lat, BASES.cavalcante.lng)
        };
    }

    attr.googlePhotosUrl = getGooglePhotosUrl(attr.name);

    // Manual overrides for accuracy
    if (attr.id === 'jardim-maytrea') {
        attr.distances = { altoParaiso: 22, saoJorge: 17, cavalcante: 109 };
    }
    if (attr.id === 'santa-barbara') {
        attr.distances = { altoParaiso: 117, saoJorge: 153, cavalcante: 27 };
    }
    if (attr.id === 'capivara') {
        attr.distances = { altoParaiso: 115, saoJorge: 151, cavalcante: 25 };
    }
});

const newArrayStr = JSON.stringify(attractions, null, 4);
const newContent = content.replace(/const ATTRACTIONS = \[[\s\S]*?\];/, `const ATTRACTIONS = ${newArrayStr};`);

fs.writeFileSync(dataPath, newContent, 'utf8');
console.log("Successfully enriched attractions data.");
